File: contests/longest_increasing_subsequence.py
class Solution:
    def longest_increasing_subsequence(self):
        nums = self.get_nums()
        longest = [1]*(len(nums))
        #[1, 2, 3, 4]
        # longest[2] = 2
        #longest[1] = 1
        #longest[3] = max(longest[2] + 1, longest[1] + 1)
        #longest[4] = max(longest[3] + 1, longest[2] + 1, longest[1] + 1) 
        #longest[4] = 1 + max(longest[3], longest[2], longest[1])
        #
        for index in range(len(nums)):
            for current in range(index):
                if nums[current] < nums[index]:
                    longest[index] = max(longest[index], 1 + longest[current])
            
        return print(max(longest))
    
    def get_nums(self):
        return list(map(int, input().split()))
    
    
    
solution = Solution()
solution.longest_increasing_subsequence()

File: longest_increasing_subsequence.cpp
using namespace std;
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n = nums.size();
        vector<int> dp(n + 1, 1);
        for (int index = 1; index < n; index++){
            for (int curr = 0; curr < index; curr++){
                if (nums[index] > nums[curr]){
                    dp[index] = max(dp[index], 1 + dp[curr]);
                }
            }
        }
        return *max_element(dp.begin(), dp.end());
        
    }
};